packages/next-app/src/app/stableswapsimulator/(components)/SwapSimulator.tsx

```diff
@@ -0,0 +1,258 @@
+                render={({ field: { onChange, value, ref } }) => (
+                  <Select onValueChange={onChange} value={value} ref={ref}>
+                    {tokensSymbol.map((tokenSymbol, index) => (
+                      <SelectItem key={tokenSymbol} value={index.toString()}>
+                        {tokenSymbol}
+                      </SelectItem>
+                    ))}
+                  </Select>
+                )}
+              />
+            </div>
+          </div>
+          <div className="flex flex-col">
+            <label className="mb-2 block text-sm text-slate12">Swap Type</label>
+            <Controller
+              control={control}
+              name="swapType"
+              defaultValue="Exact In"
+              render={({ field: { onChange, value, ref } }) => (
+                <Select onValueChange={onChange} value={value} ref={ref}>
+                  {swapTypes.map((swapType) => (
+                    <SelectItem key={swapType} value={swapType}>
+                      {swapType}
+                    </SelectItem>
+                  ))}
+                </Select>
+              )}
+            />
+          </div>
+          <Input
+            {...register("amount")}
+            label="Amount"
+            defaultValue={(initialData?.tokens[0]?.balance / 10).toFixed(2)}
+          />
+        </div>
+        <div className="flex flex-col w-1/3 gap-y-2">
+          <label className="mb-2 block text-md text-slate12 underline decoration-blue9 decoration-4">
+            Initial simulation
+          </label>
+          <SimulationResult {...initialResult} />
+        </div>
+        <div className="flex flex-col w-1/3 gap-y-2">
+          <label className="mb-2 block text-md text-slate12 underline decoration-amber9 decoration-4">
+            Custom simulation
+          </label>
+          <div className="flex flex-col gap-y-2">
+            <SimulationResult {...customResult} />
+          </div>
+        </div>
+      </div>
+    </div>
+  );
+}
+
+function calculateSimulation({
+  data,
+  swapType,
+  amount,
+  tokenInIndex,
+  tokenOutIndex,
+}: {
+  data: AnalysisData;
+  swapType: string;
+  amount: number;
+  tokenInIndex: number;
+  tokenOutIndex: number;
+}) {
+  if (
+    !data?.swapFee ||
+    !data?.ampFactor ||
+    !data?.tokens ||
+    !amount ||
+    typeof tokenInIndex == "undefined" ||
+    typeof tokenOutIndex == "undefined" ||
+    tokenInIndex == tokenOutIndex
+  ) {
+    return;
+  }
+
+  const poolPairData = MetaStableMath.preparePoolPairData({
+    indexIn: tokenInIndex,
+    indexOut: tokenOutIndex,
+    swapFee: data?.swapFee,
+    balances: data?.tokens.map((token) => token.balance),
+    amp: data?.ampFactor,
+    rates: data?.tokens.map((token) => token.rate),
+    decimals: data?.tokens.map((token) => token.decimal),
+  });
+
+  const OldBigNumberAmount = MetaStableMath.numberToOldBigNumber(
+    amount
+  ) as OldBigNumber;
+
+  let amountIn = OldBigNumberAmount;
```

Comment:
We are using `let` because you want to reassign this variable without causing a re-render? I as this because I feel we avoid using `let` but it might be the behavior we want in this case